### PR TITLE
Generate labels/annotations for Tekton results earlier

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_resolve_remote_source.py [hunk 1]
```diff
@@ -63,6 +65,22 @@ def json_filename(cls, name: Optional[str]):
             return REMOTE_SOURCE_JSON_FILENAME
 
 
+def transform_results(results):
+    if not results or not isinstance(results, list):
+        return
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_resolve_remote_source.py:70]: This check should be unnecessary, if a plugin returns `None`, all the metadata decorators exit without doing anything

File: atomic_reactor/plugins/pre_resolve_remote_source.py [hunk 2]
```diff
@@ -63,6 +65,22 @@ def json_filename(cls, name: Optional[str]):
...
+
+    remote_sources_annotations = [
+        {"name": remote_source["name"], "url": remote_source["url"]}
+        for remote_source in results
+    ]
+
+    if not remote_sources_annotations:
+        return
+
+    return json.dumps(remote_sources_annotations)
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_resolve_remote_source.py:80]: IMO we don't want to JSON-dump yet. The annotations returned from plugins should be Python objects (but compatible with JSON), the dumping to JSON happens automatically in the WorkflowData code and in the store_metadata plugin

File: atomic_reactor/plugins/pre_bump_release.py [hunk 3]
```diff
@@ -16,11 +16,13 @@
 from atomic_reactor.constants import (PLUGIN_BUMP_RELEASE_KEY, PROG, KOJI_RESERVE_MAX_RETRIES,
                                       KOJI_RESERVE_RETRY_DELAY)
 from atomic_reactor.config import get_koji_session
+from atomic_reactor.metadata import label
 from atomic_reactor.util import is_scratch_build
 from koji import GenericError
 import koji
 
 
+@label('koji-build-id')
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_bump_release.py:25]: We probably don't need this anymore, @tkdchen's #1824 is close to done so we will have the label in the postbuild task

File: tests/plugins/test_tag_and_push.py [hunk 4]
```diff
@@ -81,6 +81,26 @@
...
+
+    floating_repositories = []
+    for image in tag_conf.floating_images:
+        floating_repositories.append(image.to_str())
+
+    return {
+        "primary": primary_repositories,
+        "unique": unique_repositories,
+        "floating": floating_repositories,
+    }
```
Comment (chmeliik) [inline tests/plugins/test_tag_and_push.py:101]: Would it be possible to hardcode expected value of the annotation? I'm not a big fan of unit tests that duplicate the business logic

**Comment by chmeliik** [comment]
I'll mention that this could have been 3 separate commits (one for `remote_sources`, one for `repositories`, one for `koji-build-id`), but that's OK now

File: tests/plugins/test_tag_and_push.py (hunk 4 above)
Comment (rcerven) [reply tests/plugins/test_tag_and_push.py:101]: well tests in this plugin are skipped apart from one, so I can't really test it. So hardcoding can be done later

File: atomic_reactor/plugins/pre_bump_release.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_bump_release.py]: ok well removed

File: tests/plugins/test_tag_and_push.py (hunk 4 above)
Comment (chmeliik) [reply tests/plugins/test_tag_and_push.py:101]: Oh, does that test not produce the annotation? But we still use this plugin for source containers, and we need that annotation for source containers

Comment (rcerven) [reply tests/plugins/test_tag_and_push.py:101]: one test which is enabled produces annotation, others are disabled

**Comment by chmeliik** [comment]
> (as long as the commit message is descriptive enough)

@rcerven hint hint :)
(would be great to have an explanation why we need this)

**Review by chmeliik — APPROVED** [review]
LGTM if CI passes :+1:

**Review by hjmodi — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated

**Pull request merged**